File: 0665-Non-decreasing Array/0665-Non-decreasing Array.cpp
class Solution {
public:
    bool checkPossibility(vector<int>& nums) {
        int n = nums.size();
        int count = 0;
        for (int i = 1; i < n; ++i)
        {
            if (nums[i] < nums[i - 1])
            {
                ++count;
                if (count > 1)
                {
                    return false;
                }
                if (i > 1 && nums[i - 2] > nums[i] && i < n - 1 && nums[i + 1] < nums[i - 1] )
                {
                    return false;
                }
            }
        }
        
        return true;
    }
};

File: 0601-0700/0665-Non-decreasing Array/0665-Non-decreasing Array.py
class Solution:
    def checkPossibility(self, nums: List[int]) -> bool:
        OneDec = False
        for i in range(1, len(nums)):
            if nums[i] < nums[i - 1]:
                if OneDec:
                    return False
                OneDec = True
                if i > 1 and nums[i - 2] > nums[i] and i < len(nums) - 1 and nums[i - 1] > nums[i + 1]:
                    return False
        return True
